### ISSUE TITLE
Wrong line number and/or index is reported.

### ISSUE THREAD

**Issue by crimsonwoods** [realm/SwiftLint#213]
Wrong location of detected violation is reported when a file contains Japanese comment or string literal.

Reproducible input:
```Swift
import Foundation

class Sample {
    func sample_for_reproduction(param: AnyObject?) -> Int {
        // 日本語コメント <- Japanese
        return param as! Int // force cast violation
    }   
}
```

I think that the cause of this issue is misuse of `offset` parameter supplied to `Location` class, maybe.
In `NSString.lineAndCharacterForByteOffset`, `offset` is being used as a byte-offset. On the other hand, the offset that is being used to instantiate new `Location` object is a character-offset.

**Comment by jpsim** [comment]
Thanks for reporting this. I can confirm your assessment of the issue.

**Comment by crimsonwoods** [comment]
I wrote a patch like below to fix this issue. but I don't know what the patch is good or not.

```diff
--- a/Source/SwiftLintFramework/Extensions/String+SwiftLint.swift
+++ b/Source/SwiftLintFramework/Extensions/String+SwiftLint.swift
@@ -64,18 +64,17 @@ extension String {
 
 extension NSString {
     public func lineAndCharacterForByteOffset(offset: Int) -> (line: Int, character: Int)? {
-        return byteRangeToNSRange(start: offset, length: 0).flatMap { range in
-            var numberOfLines = 0, index = 0, lineRangeStart = 0, previousIndex = 0
-            while index < length {
-                numberOfLines++
-                if index > range.location {
-                    break
-                }
-                lineRangeStart = numberOfLines
-                previousIndex = index
-                index = NSMaxRange(lineRangeForRange(NSRange(location: index, length: 1)))
+        let range = NSRange(location: offset, length: 0)
+        var numberOfLines = 0, index = 0, lineRangeStart = 0, previousIndex = 0
+        while index < length {
+            numberOfLines++
+            if index > range.location {
+                break
             }
-            return (lineRangeStart, range.location - previousIndex + 1)
+            lineRangeStart = numberOfLines
+            previousIndex = index
+            index = NSMaxRange(lineRangeForRange(NSRange(location: index, length: 1)))
         }
+        return (lineRangeStart, range.location - previousIndex + 1)
     }
 }
```

**Comment by jpsim** [comment]
That seems right to me, @crimsonwoods! I'd appreciate a PR and ideally some unit tests with your patch. Would you be able to do that?

**Comment by crimsonwoods** [comment]
I'll do that.

**Comment by jpsim** [comment]
:smile:
